### PR TITLE
Allow returning nil from PhySL blocks

### LINKED ISSUE #432: Cannot return nil in PhySL
We should be able to return `nil` just like any other value in PhySL. However, it does not work.
e.g.
```scheme
define(f, block(
    nil
))
cout(f())
```
Output:
```
physl: exception caught:
abandoning not ready shared state: HPX(broken_promise)
```

### PR DESCRIPTION
**Pull request by parsa**
Fixes #432.

### CODE REVIEW

File: tests/regressions/execution_tree/allow_return_nil_432.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#include <string>
+#include <sstream>
+
+std::string const code = R"(
+    define(f, block(
+        nil
+    ))
+    cout(f())
+)";
+
```
Comment (hkaiser) [inline tests/regressions/execution_tree/allow_return_nil_432.cpp]: Could you please add another test where nil is not the only statement inside the block, please?

Comment (parsa) [reply tests/regressions/execution_tree/allow_return_nil_432.cpp]: Done
